### PR TITLE
app: Fix electron opening browser on page reload

### PR DESCRIPTION
**Pull request by sniok**
Currently in the app on windows when application triggers page reload it opens a browser.

The reason is difference in forward and back slashes in the url. 
in main.ts startUrl has forward slashes
but when page reloads, it requests a link with backward slashes. 

Since windows handles forward slashes just fine, I added a replacement that changes to forward slashes.

Fixes https://github.com/headlamp-k8s/headlamp/issues/1990

### CODE REVIEW

**Comment by sniok** [comment]
Oh and the code that acutally opened the browser is in main.ts:574
```typescript
    mainWindow.webContents.on('will-navigate', (event, url) => {
      if (url.startsWith(startUrl)) {
        return;
      }
      event.preventDefault();
      shell.openExternal(url);
    });
```

File: app/electron/main.ts [hunk 1]
```diff
@@ -515,7 +515,8 @@ function startElecron() {
         query: {
           backendToken: backendToken,
         },
-      });
+      })
+    ).replace(/\\/g, '/');
```
Comment (joaquimrocha) [inline app/electron/main.ts:519]: Can you add a comment explaining why this is needed?

**Review by joaquimrocha — CHANGES_REQUESTED** [review]

File: app/electron/main.ts (hunk 1 above)
Comment (sniok) [reply app/electron/main.ts]: done

**Review by joaquimrocha — APPROVED** [review]

**Review by illume — APPROVED** [review]
👍

**Pull request merged**